### PR TITLE
Add parameters and metadata to New-AzureRmPolicyAssignment and Set

### PR DESCRIPTION
**Pull request by vivsriaus**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs [hunk 1]
```diff
@@ -127,6 +127,13 @@ public class NewAzurePolicyAssignmentCmdlet : PolicyCmdletBase, IDynamicParamete
         [ValidateNotNullOrEmpty]
         public Hashtable Sku { get; set; }
 
+        /// <summary>
+        /// Gets or sets the policy assignment metadata parameter
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The metadata for policy assignment. This can either be a path to a file name containing the metadata, or the metadata as string.")]
+        [ValidateNotNullOrEmpty]
+        public string Metadata { get; set; }
```
Comment (MiYanni) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs]: Is there a reason this new property is not included in `Get-AzureRmPolicyAssignment`?
Additionally, what is the actual type of Metadata (meaning, how is it represented beyond a string, or is that basically user-defined)?

**Comment by maddieclayton** [comment]
@vivsriaus Ping on the comments above.

**Comment by vivsriaus** [comment]
@maddieclayton we'll address the comments next week. Please keep this PR opened

**Comment by maddieclayton** [comment]
Looks like I accidentally deleted the comments, updated here:

Please update the changelog here: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Resources/ChangeLog.md
Please add covering tests
Please update the help files: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md

**Comment by maddieclayton** [comment]
@vivsriaus Ping

**Comment by maddieclayton** [comment]
Closing for inactivity.  Feel free to reopen the PR once you have addressed the comments above.

**Pull request closed without merging**